src/main.rs

```diff
@@ -89,6 +89,7 @@ struct MainWinState {
     dwrite_factory: directwrite::Factory,
     render_target: Option<RenderTarget>,
     resources: Option<Resources>,
+    filename: Option<String>,
```

Comment:
Maybe it doesn't make sense to store the `filename` here but instead should be pulled in from the backend based on the view ID in the save operation? However it could also be used for window/tab title and I in that case maybe it's better to keep it locally as opposed to talking to the backend for it on each file / tab change? (Not that there's any tabs yet…). Dunno.

Reply:
There's definitely going to need to be a per-tab state object. This seems like as good a place as any for now though.

Basing it on the view id doesn't seem like a good idea. In the mac front-end, the filename is also managed by the front-end (actually as a url in the NSDocument infrastructure).